app/src/ui/multi-commit-operation/conflicts-dialog.tsx

```diff
@@ -0,0 +1,229 @@
+  readonly userHasResolvedConflicts?: boolean
+  readonly resolvedExternalEditor: string | null
+  readonly ourBranch: string
+  /* `undefined` when we didn't know the branch at the beginning of this flow */
+  readonly theirBranch?: string
+  readonly manualResolutions: Map<string, ManualConflictResolution>
+  readonly headerTitle: string
+  readonly submitButton: string
+  readonly abortButton: string
+  readonly onSubmit: (
+    dispatcher: Dispatcher,
+    repository: Repository,
+    workingDirectory: WorkingDirectoryStatus,
+    ourBranch: string,
+    theirBranch: string
+  ) => void
+  readonly onAbort: (
+    workingDirectory: WorkingDirectoryStatus,
+    manualResolutions: Map<string, ManualConflictResolution>
+  ) => void
+  readonly onDismissed: () => void
+  readonly openFileInExternalEditor: (path: string) => void
+  readonly openRepositoryInShell: (repository: Repository) => void
+  readonly someConflictsHaveBeenResolved: () => void
+}
+
+interface IConflictsDialogState {
+  readonly isCommitting: boolean
+  readonly isAborting: boolean
+}
+
+/**
+ * Modal to tell the user their encountered conflicts
+ * - To be used generically with conflicts encountered by numerous operations
+ *   such as merging, rebasing, cherry-picking, squashing, reordering, etc.
+ */
+export class ConflictsDialog extends React.Component<
+  IConflictsDialogProps,
+  IConflictsDialogState
+> {
+  public constructor(props: IConflictsDialogProps) {
+    super(props)
+    this.state = {
+      isCommitting: false,
+      isAborting: false,
+    }
+  }
+
+  /**
+   *  Provides us ability to track if user has resolved at least one conflict in
+   *  this operation
+   */
+  public componentWillUnmount() {
+    const {
+      workingDirectory,
+      userHasResolvedConflicts,
+      manualResolutions,
+      someConflictsHaveBeenResolved,
+    } = this.props
+
+    // skip this work once we know conflicts have been resolved
+    if (userHasResolvedConflicts) {
+      return
+    }
+
+    const resolvedConflicts = getResolvedFiles(
+      workingDirectory,
+      manualResolutions
+    )
+
+    if (resolvedConflicts.length > 0) {
+      someConflictsHaveBeenResolved()
+    }
+  }
+
+  /**
+   *  Invokes submit callback and dismisses modal
+   */
+  private onSubmit = async () => {
+    this.setState({ isCommitting: true })
+
+    const {
+      dispatcher,
+      repository,
+      workingDirectory,
+      ourBranch,
+      theirBranch,
+      onSubmit,
+      onDismissed,
+    } = this.props
+
+    onSubmit(dispatcher, repository, workingDirectory, ourBranch, theirBranch)
+    onDismissed()
+  }
+
+  /**
+   *  Invokes abort callback and dismisses modal
+   */
+  private onAbort = async (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.preventDefault()
+
+    const { workingDirectory, manualResolutions } = this.props
+    this.setState({ isAborting: true })
+    this.props.onAbort(workingDirectory, manualResolutions)
```

Comment:
Shouldn't this be `awaited`? Otherwise I don't really see the need for the `isAborting` state.

I've checked the current implementation for for cherry-pick and rebase, and their abort methods both are async (as one would expect) but they're not awaited O_o

Am I missing something or you think my assessment makes sense?

Reply:
:D - I did see that and was going to investigate further when I refactored to use this one.. cause I was a little confused by that as well.

Reply:
The props methods were not async.... so I figured likely a previous oversight and they would need refactored when I get to them.

Reply:
Had to fix a build error - went ahead and added the await since it is likely heading that direction.